assume_role_policy.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type TrustPolicyDocument struct {
+	Version    string      `json:",omitempty"`
+	Id         string      `json:",omitempty"`
+	Statements interface{} `json:"Statement"`
+}
+
+func SuppressEquivalentTrustPolicyDiffs(key string, old string, new string, d *schema.ResourceData) bool {
+	UnmarshalAndMarshal(&old, &new)
+	return old == new
+}
+
+//Broken into seperate function to allow for returning of errors.
+func UnmarshalAndMarshal(oldPolicy *string, newPolicy *string) error {
```

Comment:
A couple things here:

First, to make this function more reusable, why don't we instead refactor this to just act on a single policy string and then call it twice: once for each policy

Second, I noticed this function does the old-school modify-my-inputs-in-place tactic which can be weird to debug down the line. Sometimes when the inputs are really large people still do this for performance reasons but for our case I'd be fine sacrificing that tiny bit of performance to save us a few headaches down the road. Consider returning the new string containing JSON rather than modifying the input